sonar-kotlin-plugin/src/main/java/org/sonarsource/kotlin/api/FunMatcher.kt

```diff
@@ -52,18 +55,22 @@ class FunMatcherImpl(
 
     fun matches(node: KtCallExpression, bindingContext: BindingContext): Boolean {
         val call = node.getCall(bindingContext)
-        val functionDescriptor = bindingContext.get(RESOLVED_CALL, call)?.resultingDescriptor
-        return matches(functionDescriptor)
+        return preCheckArgumentCount(call) &&
+            matches(bindingContext.get(RESOLVED_CALL, call)?.resultingDescriptor)
     }
 
     fun matches(node: KtNamedFunction, bindingContext: BindingContext): Boolean {
         val functionDescriptor = bindingContext.get(BindingContext.FUNCTION, node)
         return matches(functionDescriptor)
     }
 
-    fun matches(call: Call, bindingContext: BindingContext) = matches(bindingContext.get(RESOLVED_CALL, call))
+    fun matches(call: Call, bindingContext: BindingContext) = preCheckArgumentCount(call) &&
+        matches(bindingContext.get(RESOLVED_CALL, call)?.resultingDescriptor)
 
-    fun matches(call: ResolvedCall<*>?) = matches(call?.resultingDescriptor)
+    fun matches(resolvedCall: ResolvedCall<*>?) = preCheckArgumentCount(resolvedCall?.call) &&
+        matches(resolvedCall?.resultingDescriptor)
+
+    private fun preCheckArgumentCount(call: Call?): Boolean = call == null || (call.valueArguments.size <= maxArgumentCount)
```

Comment:
You could remove the explicit return type:
```suggestion
    private fun preCheckArgumentCount(call: Call?) = call == null || (call.valueArguments.size <= maxArgumentCount)
```